spec/asyncapi.md

```diff
@@ -1510,7 +1510,7 @@ components:
 
 The Schema Object allows the definition of input and output data types.
 These types can be objects, but also primitives and arrays.
-This object is a superset of the [JSON Schema Specification Draft 07](http://json-schema.org/).
+This object is a superset of the [JSON Schema Specification Draft 07](http://json-schema.org/). This means that Schema Object can be a boolean value. `True` means that the type can be anything, while `false` means that the given schema cannot be defined.
 
```

Comment:
This is somewhat incorrect depending on how you look at it. If we are talking purely JSON Schema, then `true` means everything will be accepted, `false` means everything will be rejected. Not sure how to best describe that in this sentence though 🤔

Reply:
`then true means everything will be accepted` - `True` means that the type can be anything.
`false means everything will be rejected` - `false` means that the given schema cannot be defined.

I think that both phrases seem to have the same meaning.

The bigger problem to describe is for some parts of spec, because when we will allow making schema as boolean value (as described in spec after merging this PR), some parts, like headers` in message, MUST be of `object` type, not boolean. People may not understand exactly when they will be able to use scheme as boolean. The schema for the Channel Parameters is another example where the boolean schema doesn't make much sense, but if it's a Schema Object then they have possibility to do that...

Reply:
> `then true means everything will be accepted` - `True` means that the type can be anything.
> `false means everything will be rejected` - `false` means that the given schema cannot be defined.

Hehe, I see your point, but it is two different perspectives, the one I proposed is from a validation point of view, the other is from a definition point of view. - and as JSON Schema is validation, that is why I proposed to change the wording 🙂 

> The bigger problem to describe is for some parts of spec, ...

From my perspective, it is not a problem, as soon as you say it can be a `Schema Object` that entails that it can be a boolean. Why would you want to restrict this? 🤔 IMO, whether it is used or not, that's up to the user, not us 🙂

Reply:
I am starting to wonder if it is intentional that the schema object is not allowed to be a boolean or if that was just how it came to be? 🤔

Reply:
In spec we use usually definition point of view, in some places also validation POV, but only by purpose, we define that some parts must be in given shape or must be unique. Now I understand why you want describe `true/false` in this way, but as I read `Schema Object` section in spec I see description from definition POV, not validation 😅

Reply:
> I am starting to wonder if it is intentional that the schema object is not allowed to be a boolean or if that was just how it came to be? 🤔

Maybe @fmvilas forgot about possibility of `true/false` schemas when he described the spec 😅

Reply:
> In spec we use usually definition point of view, in some places also validation POV, but only by purpose, we define that some parts must be in given shape or must be unique. Now I understand why you want describe `true/false` in this way, but as I read `Schema Object` section in spec I see description from definition POV, not validation 😅

Yeea, and I agree we should be consistent. 

IMO it needs to be re-written to use the correct perspective, but it might become unclear for other people then... 🤷

Reply:
> IMO it needs to be re-written to use the correct perspective, but it might become unclear for other people then...

Ok, let's wait for other comments :)

Reply:
> IMO it needs to be re-written to use the correct perspective, but it might become unclear for other people then

I agree we should advocate for consistency since having two perspectives are adding confusion when reading.

> In spec we use usually definition point of view, in some places also validation POV, but only by purpose, we define that some parts must be in given shape or must be unique. Now I understand why you want describe true/false in this way, but as I read Schema Object section in spec I see description from definition POV, not validation 😅

IMHO, definition/describing POV looks more easy to understand for the first reader. Specs are made for describing, tooling around does other things such as validation.

Reply:
> Specs are made for describing, tooling around does other things such as validation.

Then what if you describe what your data (payload) should validate against? 😆

Reply:
I used the description from https://github.com/OAI/OpenAPI-Specification/pull/2609/files PR:

> The empty schema (which allows any instance to validate) MAY be represented by the `boolean` value `true` and a schema which allows no instance to validate MAY be represented by the `boolean` value `false`.

It's from validation pov, but for me it's a better than previous one.